internal/syncer/connector/notion.go

```
@@ -0,0 +1,997 @@
+		indexedPages:    map[string]bool{},
+		pagePathCache:   map[string]string{},
+	}, nil
+}
+
+// Validate validates Notion connector settings and credentials.
+func (c *NotionConnector) Validate(ctx context.Context) error {
+	if c == nil {
+		return fmt.Errorf("notion connector is nil")
+	}
+	if c.integrationKey == "" {
+		return fmt.Errorf("Notion credentials not loaded.")
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)